app/controllers/subscriptions_auth_token_controller.rb

```diff
@@ -2,13 +2,39 @@ class SubscriptionsAuthTokenController < ApplicationController
   before_action :validate_params
 
   def auth_token
+    address = params.fetch(:address)
+    topic_id = params.fetch(:topic_id)
+    frequency = params.fetch(:frequency)
+
+    token = build_token(address: address, topic_id: topic_id, frequency: frequency)
```

Comment:
Minor: the amount of parameter passing means it takes some effort to read this code. Since this is the only responsibility of the controller, could we do all or most of the work in the action method?